accepted/00078-maintenance-windows-web-calendar.md

```diff
@@ -0,0 +1,87 @@
+- Feature Name: Maintenance Windows Web Calendar
+- Start Date: 2020-12-18
+- RFC PR: (leave empty) 
+
+# Summary
+[summary]: #summary
+
+Provide the user with a calendar view that displays information about the maintenance windows associated with the calendar.
+
+# Motivation
+[motivation]: #motivation
+
+### Why are we doing this?
+Currently we display the contents of the ical file uploaded by the users in the calendar details and creation menus. Since it's difficult to draw information about the maintenance windows defined in that file from the pure ical we want to provide this information displayed as a calendar.
+
+### What use cases does it support?
+
+When choosing a maintenance schedule from a maintenance calendar the user needs to know the details of the maintenance windows contained in the calendar.
+
+Typical workflow when creating new maintenance windows:
+
+1. Head to `Schedule` -> `Maintenance Windows` -> `Calendars` -> `Create`
+2. Follow the steps to create a new calendar
+3. Go to `Schedules` to create a new maintenance schedule
+4. Use the information provided in the created calendars details view to pick a maintenance window
+5. Assign systems to the newly created maintenance schedule
+
+### What is the expected outcome?
+
+Replace the plain ical text in the calendar details and creation menu with a visual representation of the maintenance windows contained in the uploaded ical file.
+Use the created calendar view to also display information about a maintenance window in the maintenance schedule details and creation menu. 
+
+# Detailed design
+[design]: #detailed-design
+
+### UI
+
+There seem to be two solutions suited for our use case:
+1. [React Big Calendar](https://github.com/jquense/react-big-calendar)
+2. [Full Calendar](https://github.com/fullcalendar/fullcalendar)
```

Comment:
Since we have to make a choice at some point (between these two and perhaps others as @paususe also suggested) I think it would be good to start exploring pros/cons of each at this stage. Of course hacking some quick experiments is a perfectly valid way to explore.

Ideally, we would have a list of pros and cons for each alternative here. Some relevant criteria that come to my mind:
  - **risk**: how easy will it be to swap a given library with a different one later on? If many options exist and switching from one to the other is cheap, following questions are less important and a choice can be made with less detailed data, as in case of trouble we can always switch. On the other end of the spectrum, if changing our minds later will require substantial work, it becomes more important to try to get it right from the start
  - **fitness to purpose**: do solutions have all the functionality we need out of the box or not? If not, "how much" do we have to add to which solutions?
 - **integration burden**: are all solutions equally easy to integrate in the current codebase, or do some require more work than others (frontend/backend)?
 - **long-term maintenance burden**: which solutions are expected to cause less problems down the road?
   - how big is the codebase (we will have to dig through in case of L3s)? How big is the dependency tree?
   - how active is the project? (eg. are there signs of recent activity? How big is the the community? Is the codebase de facto controlled by one single company? How receptive is the project about new issues or pull requests?)
   - how stable is the project? (do we have any idea if the API changes often, or if we can expect "drop-in" updates in the future?)
   - how good is the codebase? (does it cause shivers to read a random file? are there tests? is it documented well?)

I understand I am asking you to collect quite some data points to try to answer those questions, and that takes time. Nevertheless I think such time is well spent and will definitely pay off!

Reply:
Please also consider future/nice to have features, e. g. creation of calendars from Uyuni itself (stored in Uyuni DB or git or timezone selection (show current calendar in a different timezone).

Reply:
Thank you for your valuable feedback. This is my first time working on an rfc so I expected that there will be some rough edges. I tried out and evaluated each of the 3 options (including the one mentioned by Pau) using your suggestions and updated the rfc with my findings.